plugins/querytranslate/middleware.go

```diff
@@ -131,6 +132,114 @@ func queryTranslate(h http.HandlerFunc) http.HandlerFunc {
 			return
 		}
 
+		// validate request by permission
+		reqPermission, err := permission.FromContext(req.Context())
+		if err != nil {
+			log.Warnln(logTag, ":", err)
+		}
+		if reqPermission != nil && reqPermission.ReactiveSearchConfig != nil {
+			for _, query := range body.Query {
+
+				// valiate query DSL
+				if reqPermission.ReactiveSearchConfig.DisbaleQueryDSL != nil {
+					if *reqPermission.ReactiveSearchConfig.DisbaleQueryDSL {
+						errorMsg := "raw query DSL is disabled. Please use a stored query instead"
+						if query.DefaultQuery != nil {
+							defaultQuery := *query.DefaultQuery
+							if defaultQuery != nil {
+								storedQueryId := defaultQuery["id"]
+								if storedQueryId == nil {
+									telemetry.WriteBackErrorWithTelemetry(req, w, errorMsg, http.StatusBadRequest)
+									return
+								}
+								idAsString, ok := storedQueryId.(string)
+								if !ok || idAsString == "" {
+									telemetry.WriteBackErrorWithTelemetry(req, w, errorMsg, http.StatusBadRequest)
+									return
+								}
+							}
+						}
+						if query.CustomQuery != nil {
+							customQuery := *query.CustomQuery
+							if customQuery != nil {
+								storedQueryId := customQuery["id"]
+								if storedQueryId == nil {
+									telemetry.WriteBackErrorWithTelemetry(req, w, errorMsg, http.StatusBadRequest)
+									return
+								}
+								idAsString, ok := storedQueryId.(string)
+								if !ok || idAsString == "" {
+									telemetry.WriteBackErrorWithTelemetry(req, w, errorMsg, http.StatusBadRequest)
+									return
+								}
+							}
+						}
+					}
+				}
+
+				// validate query size
+				if reqPermission.ReactiveSearchConfig.MaxSize != nil {
+					// validate size from defaultQuery if present
+					if query.DefaultQuery != nil {
+						defaultQuery := *query.DefaultQuery
+						if defaultQuery["size"] != nil {
+							sizeAsFloat, ok := defaultQuery["size"].(float64)
```

Comment:
@bietkul Getting size as float is redundant. Size should always be coercable to int. In fact, if a size like "100.50" is passed, we should throw a 400 request.